packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -7160,13 +7160,88 @@ class KeyedSubtree extends StatelessWidget {
   Widget build(BuildContext context) => child;
 }
 
-/// A platonic widget that calls a closure to obtain its child widget.
+/// A simple widget that provides a specific [BuildContext] object to its child
+/// widget.
+///
+/// This [BuildContext] object is unique and is not the same object as
+/// the one provided by the framework in the `Widget build` method.
+///
+/// Why is this important? Take the following example, where we would like to
+/// show a [SnackBar] widget to a user when they tap a button:
+///
+/// {@tool sample --template=stateless_widget_material}
+/// Example of Scaffold.of With No Builder
+/// ```dart
+///  Widget build(BuildContext context) {
+///    return Scaffold(
+///      appBar: AppBar(
+///        title: Text('Scaffold.of Example'),
+///      ),
+///      body: Center(
+///        child: RaisedButton(
+///          child: Text('Show a SnackBar'),
+///          onPressed: () {
+///            Scaffold.of(context).showSnackBar(
+///              SnackBar(
+///                content: Text('Have a snack!'),
+///              ),
+///            );
+///          },
+///        ),
+///      ),
+///    );
+///  }
+/// ```
+/// {@end-tool}
```

Comment:
Fair enough

Reply:
Also, when using the template, this will be presented as a runnable app in the documentation, so we wouldn't want it to be a bad sample that crashes. :) 
Maybe you can use the `{@tool snippet` for this one, it won't be presented as an executable, and you can expand the discussion to point out why this won't work.